Unable to use quartz as scheduler for ScheduledPollConsumer in native mode

Components that leverage `ScheduledPollConsumer` allow you to specify the scheduler to use. When using quartz, there are failures at runtime in native mode:

```
2020-09-01 09:48:00,000 ERROR [org.qua.cor.ErrorLogger] (DefaultQuartzScheduler-camel-quarkus-xml_QuartzSchedulerThread) An error occured instantiating job to be executed. job= 'DEFAULT.6da64b5bd2ee-ba5b34dc-56b8-4169-bf94-a1bae35a1df5': org.quartz.SchedulerException: Problem instantiating class 'org.apache.camel.pollconsumer.quartz.QuartzScheduledPollConsumerJob' [See nested exception: java.lang.InstantiationException: Type `org.apache.camel.pollconsumer.quartz.QuartzScheduledPollConsumerJob` can not be instantiated reflectively as it does not have a no-parameter constructor or the no-parameter constructor has not been added explicitly to the native image.]
	at org.quartz.simpl.SimpleJobFactory.newJob(SimpleJobFactory.java:58)
	at org.quartz.simpl.PropertySettingJobFactory.newJob(PropertySettingJobFactory.java:69)
	at org.quartz.core.JobRunShell.initialize(JobRunShell.java:127)
	at org.quartz.core.QuartzSchedulerThread.run(QuartzSchedulerThread.java:392)
	at com.oracle.svm.core.thread.JavaThreads.threadStartRoutine(JavaThreads.java:517)
	at com.oracle.svm.core.posix.thread.PosixJavaThreads.pthreadStartRoutine(PosixJavaThreads.java:193)
Caused by: java.lang.InstantiationException: Type `org.apache.camel.pollconsumer.quartz.QuartzScheduledPollConsumerJob` can not be instantiated reflectively as it does not have a no-parameter constructor or the no-parameter constructor has not been added explicitly to the native image.
	at java.lang.Class.newInstance(DynamicHub.java:830)
	at org.quartz.simpl.SimpleJobFactory.newJob(SimpleJobFactory.java:56)
	... 5 more
```

--- a/extensions/quartz/deployment/src/main/java/org/apache/camel/quarkus/component/quartz/deployment/QuartzProcessor.java
+++ b/extensions/quartz/deployment/src/main/java/org/apache/camel/quarkus/component/quartz/deployment/QuartzProcessor.java
@@ -24,6 +24,11 @@ import io.quarkus.deployment.builditem.nativeimage.ReflectiveClassBuildItem;
 class QuartzProcessor {
 
     private static final String FEATURE = "camel-quartz";
+    private static final String[] QUARTZ_JOB_CLASSES = new String[] {
+            "org.apache.camel.component.quartz.CamelJob",
+            "org.apache.camel.component.quartz.StatefulCamelJob",
+            "org.apache.camel.pollconsumer.quartz.QuartzScheduledPollConsumerJob"
+    };
 
     @BuildStep
     FeatureBuildItem feature() {
@@ -37,6 +42,6 @@ class QuartzProcessor {
 
     @BuildStep
     ReflectiveClassBuildItem registerForReflection() {
-        return new ReflectiveClassBuildItem(false, false, "org.apache.camel.component.quartz.CamelJob");
+        return new ReflectiveClassBuildItem(false, false, QUARTZ_JOB_CLASSES);
     }
 }

--- a/integration-tests/file/pom.xml
+++ b/integration-tests/file/pom.xml
@@ -36,7 +36,7 @@
         <!-- explicit dependencies of this module in the Maven sense, although they are required by the Quarkus Maven plugin. -->
         <!-- Please update the rule whenever you change the dependencies of this module by running -->
         <!--     mvn process-resources -Pformat    from the root directory -->
-        <mvnd.builder.rule>camel-quarkus-file-deployment,camel-quarkus-file-watch-deployment,camel-quarkus-support-policy-deployment</mvnd.builder.rule>
+        <mvnd.builder.rule>camel-quarkus-file-deployment,camel-quarkus-file-watch-deployment,camel-quarkus-quartz-deployment,camel-quarkus-support-policy-deployment</mvnd.builder.rule>
     </properties>
 
     <dependencies>
@@ -47,6 +47,10 @@
         <dependency>
             <groupId>org.apache.camel.quarkus</groupId>
             <artifactId>camel-quarkus-file-watch</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.camel.quarkus</groupId>
+            <artifactId>camel-quarkus-quartz</artifactId>
         </dependency>
         <dependency>
             <groupId>io.quarkus</groupId>

--- a/integration-tests/file/src/main/java/org/apache/camel/quarkus/component/file/it/FileRoutes.java
+++ b/integration-tests/file/src/main/java/org/apache/camel/quarkus/component/file/it/FileRoutes.java
@@ -40,5 +40,7 @@ public class FileRoutes extends RouteBuilder {
                 + "readLockTimeout=5000")
                         .to("file://target/" + READ_LOCK_OUT);
 
+        from("file://target/quartz?scheduler=quartz&scheduler.cron=0/1+*+*+*+*+?&repeatCount=0")
+                .to("file://target/quartz/out");
     }
 }

--- a/integration-tests/file/src/test/java/org/apache/camel/quarkus/component/file/it/FileTest.java
+++ b/integration-tests/file/src/test/java/org/apache/camel/quarkus/component/file/it/FileTest.java
@@ -28,9 +28,9 @@ import io.restassured.RestAssured;
 import io.restassured.http.ContentType;
 import io.restassured.path.json.JsonPath;
 import io.restassured.response.ValidatableResponse;
-import org.awaitility.Awaitility;
 import org.junit.jupiter.api.Test;
 
+import static org.awaitility.Awaitility.await;
 import static org.hamcrest.core.IsEqual.equalTo;
 
 @QuarkusTest
@@ -105,8 +105,32 @@ class FileTest {
                 .statusCode(204);
     }
 
+    @Test
+    public void quartzSchedulerFilePollingConsumer() throws InterruptedException {
+        String fileName = RestAssured.given()
+                .contentType(ContentType.TEXT)
+                .body(FILE_BODY)
+                .post("/file/create/quartz")
+                .then()
+                .statusCode(201)
+                .extract()
+                .body()
+                .asString();
+
+        String targetFileName = Paths.get(fileName).toFile().getName();
+        await().atMost(10, TimeUnit.SECONDS).until(() -> {
+            return Files.exists(Paths.get("target/quartz/out", targetFileName));
+        });
+
+        RestAssured
+                .get("/file/get/{folder}/{name}", "quartz/out", targetFileName)
+                .then()
+                .statusCode(200)
+                .body(equalTo(FILE_BODY));
+    }
+
     private static void awaitEvent(final Path dir, final Path file, final String type) {
-        Awaitility.await()
+        await()
                 .pollInterval(10, TimeUnit.MILLISECONDS)
                 .atMost(10, TimeUnit.SECONDS)
                 .until(() -> {

